services/paymaster/src/api.rs

```diff
@@ -358,22 +368,21 @@ async fn sponsor_and_send_handler(
 
         Ok(SponsorAndSendResponse::Confirm(confirmation_result))
     } else {
-        let signature = state
-            .chain_index
-            .rpc
-            .send_transaction_with_config(
-                &transaction,
-                RpcSendTransactionConfig {
-                    skip_preflight: true,
-                    ..RpcSendTransactionConfig::default()
-                },
+        let signature = send_transaction(
```

Comment:
fyi: we are trying to get rid of this branch. New versions of the typescript sdk use the version where the paymaster waits for confirmation